src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java

```diff
@@ -0,0 +1,248 @@
+		while(!(I18n.translateToLocal(key)).equals(key)) {
+			key = "tropicraft.loading."+cat+"."+count++;
+		}
+		
+		screenTitle = "tropicraft.loading."+cat+"."+String.valueOf(rand.nextInt(count+1)+1);
+		if(cat.equals("knowledge") || cat.equals("hint")) {
+			String prepend = I18n.translateToLocal("tropicraft.loading."+cat+".pre");
+			screenTitle = I18n.translateToLocalFormatted(screenTitle, prepend);
+		}else {
+			screenTitle = I18n.translateToLocal(screenTitle);
+
+		}
+		
+		// Pick random background
+		screenBackground = BACKGROUNDS[rand.nextInt(BACKGROUNDS.length)];
+	}
+	
+	
+	@SubscribeEvent
+	public void onRenderOverlay(RenderTickEvent evt) {
+		if(evt.type != TickEvent.Type.RENDER) return;
+		if(evt.phase != Phase.END) return;	
+	
+		if(mc.currentScreen instanceof GuiDownloadTerrain) {
+			if(this.screenReassign) {
+				this.assignScreenContent();
+				this.screenReassign = false;
+			}
+			if(mc.world != null) {
+				if(lastWorldUpdateTick != mc.world.getTotalWorldTime() % 240000) {
+					
+					lastWorldUpdateTick = mc.world.getTotalWorldTime() % 240000;
+					
+					if(lastWorldUpdateTick % 200 == 0) {
+						assignScreenContent();
+					}
+				}
+			}
+			if(mc.player != null && mc.player.dimension == TropicraftWorldUtils.TROPICS_DIMENSION_ID) {
+				if(screenEntities == null && mc.world != null){
+					assignScreenContent();
+				}
+				FontRenderer f = mc.fontRendererObj;
+				ScaledResolution sr = new ScaledResolution(mc);
+				
+				drawBackground(sr.getScaledWidth(), sr.getScaledHeight());
+				if(screenEntities != null) {
+					drawScreenEntity(screenEntities.getLeft(), sr.getScaledWidth()/2 - 120, (sr.getScaledHeight()/2)+60, 50, -90+(int)(this.lastWorldUpdateTick*2), 20);
+					drawScreenEntity(screenEntities.getRight(), sr.getScaledWidth()/2 + 120, (sr.getScaledHeight()/2)+60, 50, 90-(int)(this.lastWorldUpdateTick*2), 20);
+					
+					drawScreenItemStack(screenItems.getLeft(), (sr.getScaledWidth()/2)-8-24, (sr.getScaledHeight()/2)+8, 1.5f);
+					drawScreenItemStack(screenItems.getMiddle(), (sr.getScaledWidth()/2)-12, (sr.getScaledHeight()/2)+8, 1.5f);
+					drawScreenItemStack(screenItems.getRight(), (sr.getScaledWidth()/2)-8+16, (sr.getScaledHeight()/2)+8, 1.5f);
+				}
+				String msg = I18n.translateToLocal("tropicraft.loading.title");
+				GlStateManager.pushMatrix();
+				GlStateManager.translate((sr.getScaledWidth())/2 - (f.getStringWidth(msg)/2), (sr.getScaledHeight()/4) - 30, 0);
+				GlStateManager.scale(2f, 2f, 1f);
+				GlStateManager.translate(-(f.getStringWidth(msg)/4), 0, 0);
+				f.drawStringWithShadow(msg, 0, 0, 0x64fc97);
+				GlStateManager.popMatrix();
+		
+				String subtitle = screenTitle;
+				
+				f.drawStringWithShadow(subtitle, (sr.getScaledWidth())/2 - (f.getStringWidth(subtitle)/2), ((sr.getScaledHeight()/1.2f)) + 5, 0xf7bf56);
+						
+				GlStateManager.color(1f, 1f, 1f, 1f);
+				
+						
+			}
+		}else {
+			this.screenReassign = true;
+		}
+	}
+	
+	public ItemStack stack(Item i) {
```

Comment:
I don't see the point of these methods

Reply:
saves a 'new item' being typed every time i need an itemstack, over a lifetime i could save a few hours just by not typing new ItemStack